core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/EnumOptionSourceResolver.java

```diff
@@ -0,0 +1,171 @@
+                try {
+                    clazz = (Class<? extends Enum<?>>) bundle.loadClass(path);
+                    if (clazz != null) {
+                        break;
+                    }
+                } catch (ClassNotFoundException ignore) {
+                }
+            }
+        }
+
+        if (clazz == null) {
+            LOG.error("EnumOptionSourceResolver::resolve - value isn't a valid class name: " + path);
+            return null;
+        }
+
+        if (!clazz.isEnum()) {
+            LOG.error("EnumOptionSourceResolver::resolve - class " + clazz.getName() + " is not an Enum");
+            return null;
+        }
+
+        if (clazz.getEnumConstants().length == 0) {
+            LOG.error("EnumOptionSourceResolver::resolve - Enum class " + clazz.getName() + " doesn't have constants");
+            return null;
+        }
+
+        List<Resource> children = new ArrayList<>();
+
+        for (Enum<?> enm : clazz.getEnumConstants()) {
+            Map<String, Object> map = new HashMap<>();
+
+            map.put(JcrConstants.JCR_TITLE, enm.name());
+            map.put(VALUE_ATTRIBUTE_KEY, enm.toString());
+
+            if (pathParameters.getTextMember() != null) {
+                map.put(pathParameters.getTextMember(), getEnumMethodInvokeResult(pathParameters.getTextMember(), enm));
+            }
+
+            if (pathParameters.getValueMember() != null) {
+                map.put(pathParameters.getValueMember(),
+                    getEnumMethodInvokeResult(pathParameters.getValueMember(), enm));
+            }
+
+            if (pathParameters.getAttributeMembers() != null) {
+                for (String methodName : pathParameters.getAttributeMembers()) {
+                    map.put(methodName, getEnumMethodInvokeResult(methodName, enm));
+                }
+            }
+
+            children.add(new ValueMapResource(request.getResourceResolver(),
+                StringUtils.EMPTY,
+                StringUtils.EMPTY,
+                new ValueMapDecorator(map)));
+        }
+
+        return new ValueMapResource(request.getResourceResolver(),
+            StringUtils.EMPTY,
+            StringUtils.EMPTY,
+            new ValueMapDecorator(Collections.emptyMap()),
+            children);
+    }
+
+    /**
+     * Called by {@link EnumOptionSourceResolver#resolve(SlingHttpServletRequest, String, PathParameters)}
+     * to retrieve a method by
+     * @param methodName from enum {@link Class}. Then it invokes on
+     * @param enm        constant and
+     * @return {@link String} value or an empty {@link String} in case of {@link Exception}
+     */
+    private String getEnumMethodInvokeResult(String methodName, Enum<?> enm) {
+        try {
+            return (String) clazz.getMethod(methodName).invoke(enm);
+        } catch (NoSuchMethodException ignore) {
+            LOG.error("EnumOptionSourceResolver::resolve - there is no method " + methodName + " at enum class "
```

Comment:
Could u use formatting for logs like in String.format? Example:  `LOG.error("EnumOptionSourceResolver::resolve - can't invoke method {0} on Enum {1}", methodName, enm.name());`